source/extensions/config_subscription/grpc/grpc_mux_impl.cc

```diff
@@ -292,6 +298,41 @@ GrpcMuxWatchPtr GrpcMuxImpl::addWatch(const std::string& type_url,
   return watch;
 }
 
+// void GrpcMuxImpl::updateMuxSource(Grpc::RawAsyncClientPtr primary_async_client,
+// Grpc::RawAsyncClientPtr failover_async_client, CustomConfigValidatorsPtr
+// custom_config_validators, BackOffStrategyPtr backoff_strategy, const
+// envoy::config::core::v3::ApiConfigSource& ads_config_source) {
+absl::Status
+GrpcMuxImpl::updateMuxSource(Grpc::RawAsyncClientPtr primary_async_client,
+                             Grpc::RawAsyncClientPtr failover_async_client,
+                             CustomConfigValidatorsPtr custom_config_validators,
+                             Stats::Scope& scope, BackOffStrategyPtr backoff_strategy,
+                             const envoy::config::core::v3::ApiConfigSource& ads_config_source) {
+  // Process the rate limit settings.
+  absl::StatusOr<RateLimitSettings> rate_limit_settings_or_error =
```

Comment:
would this result in error status if ratelimit config is not set?

Reply:
and this is because ratelimit config is required? could there be use case where user does not need ratelimit config?

Reply:
Rate-limiting is an optional field (see [rate_limit_settings field](https://github.com/envoyproxy/envoy/blob/a65a5b673e6ac0ef12a3d8c5c2cdd2be4295927b/api/envoy/config/core/v3/config_source.proto#L107)).
There are default values that are returned if the rate-limit is empty (see [RateLimitSettings struct](https://github.com/envoyproxy/envoy/blob/a65a5b673e6ac0ef12a3d8c5c2cdd2be4295927b/source/common/config/utility.h#L51) and [parseRateLimitSettings() method](https://github.com/envoyproxy/envoy/blob/a65a5b673e6ac0ef12a3d8c5c2cdd2be4295927b/source/common/config/utility.cc#L206)).

Reply:
I guess my answer should be: it is not an error to have an empty rate-limit-settings field.